src/dag-link/dagLink.js

```diff
@@ -14,13 +27,11 @@ class DAGLink {
     // assert(size, 'A link requires a size')
     //  note - links should include size, but this assert is disabled
     //  for now to maintain consistency with go-ipfs pinset
+    this.Name = name || ''
+    this.Tsize = size
+    this.Hash = new CID(cid)
 
-    Object.defineProperties(this, {
-      Name: { value: name || '', writable: false, enumerable: true },
-      Tsize: { value: size, writable: false, enumerable: true },
-      Hash: { value: new CID(cid), writable: false, enumerable: true },
-      _nameBuf: { value: null, writable: true, enumerable: false }
-    })
+    this._nameBuf = null
```

Comment:
This change probably breaks js-ipld. I need to check.

Reply:
At one point is used iterating over properties for the `tree()` call, but the values for `tree()` are hard-coded, so it won't break js-ipld.